scipy/integrate/quadpack.py

```diff
@@ -392,9 +392,8 @@ def _quad(func,a,b,args,full_output,epsabs,epsrel,limit,points):
         if infbounds != 0:
             raise ValueError("Infinity inputs cannot be used with break points.")
         else:
-            nl = len(points)
-            the_points = numpy.zeros((nl+2,), float)
-            the_points[:nl] = points
+            the_points = set(p for p in points if a < p < b)
+            the_points = numpy.array(list(the_points)+[0,0], float)
```

Comment:
Maybe write with numpy primitives?
```
points = np.asarray(points, dtype=float)
points = points[(a < points) & (points < b)]
points = np.r_[points, 0, 0]
```
starts to be faster for > 30 points.

Reply:
That doesn't get rid of the duplicates, though.  Would you use `np.unique()` instead?

Reply:
Ah yes, I meant using np.unique also.